models/articles.models.js

```diff
@@ -50,40 +51,47 @@ exports.selectCommentByArticleId = (
 exports.createComment = (article_id, newComment) => {
   const { username, body } = newComment;
 
-// check if any values are missing from the comment
+  // check if any values are missing from the comment
   if (!username || !body) {
     return Promise.reject({ status: 400, msg: "required values missing!" });
   }
 
   // check if any of the values given are invalid data
-  if (typeof username !== "string" || typeof body !==  "string") {
-    return Promise.reject({status: 400, msg:"ooops! bad request: invalid data!"})
+  if (typeof username !== "string" || typeof body !== "string") {
+    return Promise.reject({ status: 400, msg: "ooops! bad request: invalid data!" });
   }
 
-  // check if article_id is valid
+  // check if username exists
   return db
-    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .query("SELECT * FROM users WHERE username = $1", [username])
```

Comment:
It's not necessary to check the username in this way. If you built the test first you would see a different error, you wouldn't get a 200 instead of a 404. 

Have a look in the seed, the comments table is set up with a foreign key constraint. The comments.user references the users table. This means that PSQL will throw an error if you try to insert a comment with a username that doesn't exist in the users table. The error is a 23503, you'd need to add another piece of conditional logic in your PSQL error handler, and send back a 404 there. 

I feel bad because your solution works and is creative. But you've shown a checkExists for ticket 6, it would be more impressive to show your understanding of PSQL errors and foreign key constraints.

Reply:
I appreciate this is an essay, let me know if you need clarification/help!